### ISSUE TITLE
Assigning homework sets display proctor set users.

### ISSUE THREAD

**Issue by somiaj** [openwebwork/webwork2#1565]
When clicking on the 'X/Y' assigned users link on the homework set editor, the following proctor users appeared in the list:

![myscrot](https://user-images.githubusercontent.com/18082412/151309116-405454b7-67d5-48d6-8c9b-316b503e18b1.png)

These users were created while testing out proctored gateway/quizzes, and setting a set password. I assume they are used for that, and are incorrectly being displayed on this list.

**Comment by somiaj** [comment]
They also are included in the count when displaying how many total users a set is assigned to.

![myscrot](https://user-images.githubusercontent.com/18082412/151309542-0605e47d-be40-4e4c-9a99-fa4812e9382d.png)

They are the two unassigned users.

**Comment by drgrice1** [comment]
Yes, that is a minor bug that should be fixed.  The login proctor users are filtered out when displaying users in the class list editor, but they aren't filtered out here and should be.

**Comment by somiaj** [comment]
In looking into filtering these login proctor users in assignments I noticed that they are not filtered in most places (and probably should be?). The same issue as described above occurs on the achievement page where you can assign achievements to these users.

Right now the filtering is done on a per-case basis: `my @allUserIDs = grep {$_ !~ /^set_id:/} $db->listUsers;`

Would it be more appropriate to filter the users at the `$db->listUsers` level, that way by default they are not included and one doesn't need to manually filter these users out in each and every spot? And then include a `$db->listAllUsers` macro that won't filter these system users for the few places that they may need?

**Comment by drgrice1** [comment]
@somiaj:  I have been working on some improvements to many of the database queries in WeBWorK.  This is a related issue that I will take care of with that.  It don't think that it would be appropriate to filter at the `$db-listUsers` level.  Instead this should be filtered where it needs to be.

**Comment by somiaj** [comment]
@drgrice1 Thanks for the response. I was thinking adding a filter would be straight forward here so was going to write up a fix, but when looking around noticed other places that the login proctor users should be filtered, hence wondering where it would be more appropriate to filter these users. Since you seem to have this as part of some other things you are working on, I'll not continue down this path.

**Comment by drgrice1** [comment]
Looking at it closely, it seems that it would be okay to filter out the set level proctors in `listUsers`.  There is only one place that needs the result unfiltered.  That is in `DBImportExport.pm`.  It needs to export all users including the set level proctors to the export file.  The only place that this is used is in the `wwdb` script.  I am not sure if that script is used by anyone anymore.  In any case, it really doesn't need to call `listUsers` at all, and I rewrote it to directly get all user records.

I added the filter to `listUsers` for now, although it may not matter.  The real issue is that almost all places where that method is called, it should not be.  This is actually one of the main points of what I am working on.  It is inefficient and really poorly though out to make two queries to the database, the first to list all user_id's in the user table, and the second to get all records in that same table.  Instead, just get all records in the table.  That already has the list of all user_id's.  This is done with all tables in webwork.
